service_crategen/src/codegen/generator/rest_json.rs

```diff
@@ -364,11 +356,110 @@ fn generate_documentation(operation: &Operation) -> Option<String> {
+    if operation.output.is_none() {
+        return "".to_string();
     }
+
+    let shape_name = &operation.output.as_ref().unwrap().shape;
+    let output_shape = &service.shapes()[shape_name];
+
+    let mut status_code_parser = "".to_string();
+
+    for (member_name, member) in output_shape.members.as_ref().unwrap() {
+        if let Some(ref location) = member.location {
+            if location == "statusCode" {
+                if output_shape.required(member_name) {
+                    status_code_parser += &format!("result.{} = StatusCode::to_u16(&response.status);", member_name.to_snake_case());
+                } else {
+                    status_code_parser += &format!("result.{} = Some(StatusCode::to_u16(&response.status) as i64);", member_name.to_snake_case());
+                }
+            }
+        }
+    }
+
+    status_code_parser
+}
+
+
+/// Generate code to parse the http response body, either as a JSON object
+/// deserialized with serde, or as a raw payload that's assigned to one of
+/// the fields in the result object.
+///
+/// Needs to determine whether or not other fields in the result object
+/// will be set later (e.g. from headers), so the compiler won't spit out
+/// warnings about unnecessary mutability
+fn generate_body_parser(operation: &Operation, service: &Service) -> String {
+
+    if operation.output.is_none() {
+        return "let result = ();".to_string();
+    }
+
+    let shape_name = &operation.output.as_ref().unwrap().shape;
+    let output_shape = &service.shapes()[shape_name];
+
+    let mutable_result = output_shape.members
+                .as_ref()
+                .unwrap()
+                .iter()
+                .any(|(_, member)| member.location.is_some());
+
+    match output_shape.payload {
+        None => json_body_parser(shape_name, mutable_result),
+        Some(ref payload_member_name) => {
+            let payload_member_shape = &output_shape.members.as_ref().unwrap()[payload_member_name].shape;
+            let payload_shape = &service.shapes()[payload_member_shape];
+            match payload_shape.shape_type {
+                payload_type if payload_type == ShapeType::Blob ||
+                                payload_type == ShapeType::String => {
+                    payload_body_parser(payload_type, shape_name, payload_member_name, mutable_result)
+                }
+                _ => json_body_parser(shape_name, mutable_result),
+            }
+        }
+    }
+}
+
+/// Take the raw http response body and assign it to the payload field
+/// on the result object
+fn payload_body_parser(payload_type: ShapeType, output_shape: &str, payload_member: &str, mutable_result: bool) -> String {
+
+    let response_body = match payload_type {
+        ShapeType::Blob => "Some(response.body)",
+        _ => "Some(String::from_utf8_lossy(&response.body).into_owned())",
+    };
+
+    format!("
+        let {mutable} result = {output_shape}::default();
```

Comment:
Yeah, in this case I was just taking the existing generator code and massaging it a bit.  Though I suppose now is as good a time as any to make that kind of a change.